arangod/Network/Methods.cpp

```diff
@@ -376,24 +389,39 @@ FutureRes sendRequestRetry(ConnectionPool* pool, DestinationId destination,
                            arangodb::fuerte::RestVerb type, std::string path,
                            velocypack::Buffer<uint8_t> payload,
                            RequestOptions const& options,
-                           Headers headers) {
-  if (!pool || !pool->config().clusterInfo) {
-    LOG_TOPIC("59b96", ERR, Logger::COMMUNICATION)
+                           Headers headers) noexcept {
+
+  // WARNING this method shall never throw - anything!
+
+  try {
+
+    if (!pool || !pool->config().clusterInfo) {
+      LOG_TOPIC("59b96", ERR, Logger::COMMUNICATION)
         << "connection pool unavailable";
-    return futures::makeFuture(Response{destination, Error::Canceled, nullptr});
-  }
-  
-  LOG_TOPIC("2713b", DEBUG, Logger::COMMUNICATION)
+      return futures::makeFuture(Response{destination, Error::Canceled, nullptr});
+    }
+
+    LOG_TOPIC("2713b", DEBUG, Logger::COMMUNICATION)
       << "request to '" << destination
       << "' '" << fuerte::to_string(type) << " " << path << "'";
 
-  //  auto req = prepareRequest(type, path, std::move(payload), timeout, headers);
-  auto rs = std::make_shared<RequestsState>(pool, std::move(destination),
-                                            type, std::move(path),
-                                            std::move(payload),
-                                            std::move(headers), options);
-  rs->startRequest();  // will auto reference itself
-  return rs->future();
+    //  auto req = prepareRequest(type, path, std::move(payload), timeout, headers);
+    auto rs = std::make_shared<RequestsState>(pool, std::move(destination),
+                                              type, std::move(path),
+                                              std::move(payload),
+                                              std::move(headers), options);
+    rs->startRequest();  // will auto reference itself
+    return rs->future();
+
+  } catch (std::exception const& e) {
+    LOG_TOPIC("6d723", DEBUG, Logger::COMMUNICATION)
+      << "failed to send request to destination " << destination << ": " << e.what();
+  } catch (...) {
+    LOG_TOPIC("d7236", DEBUG, Logger::COMMUNICATION) << "failed to send request to destination " << destination;
+  }
+
+  return futures::makeFuture(Response{std::move(destination), Error::Canceled, nullptr});
```

Comment:
`makeFuture` can also throw exceptions, because it allocates memory. So marking this method `noexcept` is not safe.